addons/website_event/controllers/main.py

```diff
@@ -215,18 +217,25 @@ def _prepare_event_register_values(self, event, **post):
         }
 
     @http.route('/event/add_event', type='json', auth="user", methods=['POST'], website=True)
-    def add_event(self, event_name="New Event", **kwargs):
-        event = self._add_event(event_name, request.context)
+    def add_event(self, event_name, event_start, event_end, address_id, **kwargs):
+        values = {
+            'name': event_name,
+            'date_begin': event_start,
+            'date_end': event_end,
+            'address_id': address_id
+        }
+        event = self._add_event(values, request.context)
         return "/event/%s/register?enable_editor=1" % slug(event)
 
-    def _add_event(self, event_name=None, context=None, **kwargs):
-        if not event_name:
-            event_name = _("New Event")
-        date_begin = datetime.today() + timedelta(days=(14))
+    def _add_event(self, values, context=None, **kwargs):
+        date_begin = pytz.timezone(request.env.user.tz).localize(parse(values['date_begin'])).astimezone(pytz.utc).replace(tzinfo=None)
+        date_end = pytz.timezone(request.env.user.tz).localize(parse(values['date_end'])).astimezone(pytz.utc).replace(tzinfo=None)
+
         vals = {
-            'name': event_name,
-            'date_begin': fields.Date.to_string(date_begin),
-            'date_end': fields.Date.to_string((date_begin + timedelta(days=(1)))),
+            'name': values['name'] or _("New Event"),
+            'date_begin': date_begin,
+            'date_end': date_end,
+            'address_id': request.env['res.partner'].browse(int(values['address_id'])).id if values['address_id'] else request.env['res.partner'],
```

Comment:
A bit complicated, isn't it ? Either you don't trust incoming value and you could do either a search, either an exists, either you trust it and basta. But that line is a bit old fashioned (and does not filter out invalid values as browse does not perform any DB access until you try to read a field) .